### PR TITLE
Migration to ruff, and some updates

### PR DESCRIPTION
**Pull request by atztogo**
- README.rst to README.md

### CODE REVIEW

**Comment by atztogo** [comment]
I found several tests had been overrode by other tests that have the respective same function names.

I found 
- `test_xml_dielectrics(xml_parser)`: When running it, it was found to be broken. I just commented out because the code is not written so that it will pass. 
- `test_doscar_spin`: Two parameters were used and when running it, one of them failed. I just separate this test into two, `test_doscar_spin` and `test_doscar_spin_pdos`. The former passes. For the later, I changed the reference value so that the test passes.

File: parsevasp/kpoints.py [hunk 1]
```diff
@@ -321,9 +315,9 @@ def modify(self, entry, value, point_number=None):
                 self._check_points(points=value)
                 # Check that all points are in direct, if not,
                 # convert
-                for point in value:
+                for i_value, point in enumerate(value):
                     if not point.get_direct():
-                        point = self._to_direct(point)
+                        value[i_value] = self._to_direct(point)
```
Comment (atztogo) [inline parsevasp/kpoints.py:320]: I got warning from ruff. I think this was a bug, and I fixed in this way.

File: parsevasp/poscar.py [hunk 2]
```diff
@@ -158,20 +163,21 @@ def _from_dict(self):
...
-                    velocities = site.get_velocities()
-                    position = self._to_direct(position, self._poscar_dict['unitcell'])
-                    site.set_position(position)
-                    if velocities is not None:
-                        velocities = self._to_direct(velocities, self._poscar_dict['unitcell'])
-                        site.set_velocities(velocities)
-                    site.set_direct(True)
-
-    def _from_list(self, poscar):  # pylint: disable=R0915
+                sites[i_site] = Site(
```
Comment (atztogo) [inline parsevasp/poscar.py:166]: I got warning from ruff. I think this was a bug, and I fixed in this way.

**Comment by atztogo** [comment]
@zhubonan, I would like to ask you to review this PR.

I migrated to ruff. The ruff settings follow those of aiida-vasp. I added N803 for tests and N802 with respect to those of aiida-vasp.

I removed the tox setting. Then I updated `.github/workflows/ci.yml`. Now the tests run for python 3.9 to 3.12 independently without using tox. Pre-commit run in ci.yml was also removed. I hope pre-commit.ci is enough.

I found some issues on tests, which are written in my comment above. I hope those are not serious.

Ruff gave me warnings and I fixed them since those look like bugs. But I am not sure if I should fix them or not.

**Review by zhubonan — APPROVED** [review]
Looks good to me!

**Pull request merged**